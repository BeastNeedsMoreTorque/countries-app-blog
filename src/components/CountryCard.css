.country-card {
  width: 200px;
  height: 250px;
  border: 1px solid rgba(158, 158, 158, 0.5);
  box-shadow: 8px 10px 15px -6px rgba(158, 158, 158, 1);
  background: #ddd;
  margin: 1rem;
  padding: 0 1rem;
  border-radius: 4px;
  transition: all 0.5s;
  text-align: center;
}

.country-card:hover {
  transform: scale(1.05);
}

.country-info-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.country-info-container p {
  padding: 0 0.5rem;
}

.country-flag {
  width: 100px;
  height: auto;
}
