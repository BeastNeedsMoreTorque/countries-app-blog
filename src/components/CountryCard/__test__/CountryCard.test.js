import React from "react";
import CountryCard from "../CountryCard";

import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

const mockedCountry = {
  name: "Slovakia",
  population: 5000000,
  capital: "Bratislava",
  region: "Europe",
  flag: "../../assets/favourites/Canada.jpg"
};

afterEach(cleanup);

it("matches snapshot", () => {
  const card = renderer
    .create(<CountryCard country={mockedCountry} />)
    .toJSON();
  expect(card).toMatchSnapshot();
});
