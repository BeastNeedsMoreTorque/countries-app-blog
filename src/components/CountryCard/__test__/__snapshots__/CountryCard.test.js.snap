// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="country-card"
  data-testid="country-card"
>
  <h3>
    Slovakia
  </h3>
  <hr />
  <ul
    className="country-info-list"
  >
    <li
      className="country-info-item"
    >
      <h4>
        Capital:
      </h4>
      <p>
        Bratislava
      </p>
    </li>
    <li
      className="country-info-item"
    >
      <h4>
        Population:
      </h4>
      <p>
        5000000
      </p>
    </li>
    <li
      className="country-info-item"
    >
      <h4>
        Region:
      </h4>
      <p>
        Europe
      </p>
    </li>
    <li
      className="country-info-item"
    >
      <img
        alt="Country flag"
        className="country-flag"
        src="../../assets/favourites/Canada.jpg"
      />
    </li>
  </ul>
</div>
`;
